packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/util/CoroutineDispatcherFactory.kt

```diff
@@ -81,24 +83,31 @@ public sealed interface DispatcherHolder {
      */
     public val dispatcher: CoroutineDispatcher
 
+    /**
+     * Realm scheduler, we need to hold a reference to prevent being GC'ed
+     */
+    public val realmScheduler: RealmSchedulerPointer
```

Comment:
I think that we should rename the `DispatchHolder` to something that also indicates that it holds on to the scheduler. `RealmExecutionContext` or something in that direction 🤪  #naming